### PR TITLE
Raise warning instead of error when qibo version mismatch

### PR DESCRIPTION
**Pull request by marcorossi5**

### CODE REVIEW

File: src/qibo_client/qibo_client.py [hunk 1]
```diff
@@ -142,11 +142,13 @@ def check_client_server_qibo_versions(self):
         qibo_server_version = response.json()["qibo_version"]
         qibo_local_version = qibo.__version__
 
-        msg = (
-            "Local Qibo package version does not match the server one, please "
-            f"upgrade: {qibo_local_version} -> {qibo_server_version}"
-        )
-        assert qibo_local_version == qibo_server_version, msg
+        if qibo_local_version != qibo_server_version:
```
Comment (scarrazza) [inline src/qibo_client/qibo_client.py:145]: As discussed we should place a minimum version, e.g. something like:
```suggestion
        if qibo_local_version >= qibo_server_version:
```

Comment (marcorossi5) [reply src/qibo_client/qibo_client.py:145]: I'd better put `qibo_local_version <= qibo_server_version`, then raise the warning.

Comment (scarrazza) [reply src/qibo_client/qibo_client.py:145]: I think we should force local versions to be equal or larger than server, the latter when we are developing the software or releasing to pypi. If the user is using a lower version the outcome is unknown, so we should not allow executions with `local <= server`.

Comment (marcorossi5) [reply src/qibo_client/qibo_client.py:145]: Ok, the if condition should warn if `local != server` and raise error if `local < server`. Right?

Comment (marcorossi5) [reply src/qibo_client/qibo_client.py:145]: True, I was forgetting. We put `0.2.4` as the minimum version and then raise an error if there's a mismatch

Comment (scarrazza) [reply src/qibo_client/qibo_client.py:145]: Yes, exactly.

**Review by scarrazza — APPROVED** [review]

**Comment by scarrazza** [comment]
Thanks @marcorossi5, could you please fix coverage?

**Comment by marcorossi5** [comment]
It's done, hit 100% coverage.
Do we have to make a new release when merging this?

**Comment by scarrazza** [comment]
Thanks @marcorossi5, yes I will do the release if you don't plan to add other features, please let me know.

**Comment by marcorossi5** [comment]
Not for now. Do you want me to update `"0.0.3"` -> `"0.0.4"` with a new commit in this branch?

**Comment by scarrazza** [comment]
No, thanks, the next release is 0.0.3.

**Comment by marcorossi5** [comment]
ok, fine. It's ready for my

**Pull request merged**